### PR TITLE
added aws kb server

### PR DESCRIPTION
**Pull request by Skirano**
## Description
Addition of a new MCP server for AWS Knowledge Base retrieval, leveraging the Bedrock Agent Runtime to fetch information based on user queries.

## Server Details

- **Server**: aws-kb-retrieval
- **Changes to**: tools (new tool added)

## Motivation and Context
To enhance the capability of the MCP client by enabling it to retrieve information from AWS Knowledge Bases directly. This integration simplifies the process of sourcing relevant data for users.

## How Has This Been Tested?
The server has been tested with an LLM client by integrating it into the Claude Desktop setup. Various scenarios, including different queries and Knowledge Base IDs, were tested to ensure functionality.

## Breaking Changes
Users will need to update their MCP client configurations to include this new server and provide their AWS credentials.

## Types of changes

- [x] New MCP Server
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist

- [x] I have read the [MCP Protocol Documentation](https://modelcontextprotocol.io)
- [x] My server follows MCP security best practices
- [x] I have updated the server's README accordingly
- [x] I have tested this with an LLM client
- [x] My code follows the repository's style guidelines
- [x] New and existing tests pass locally
- [x] I have added appropriate error handling
- [x] I have documented all environment variables and configuration options

## Additional context
This integration requires users to set up their AWS credentials securely. The server has been optimized for performance and security, with careful consideration given to error handling and edge cases.

This form is filled out based on the AWS Knowledge Base Retrieval server you've developed. Adjustments can be made as necessary to align with your specific implementation or any additional requirements.

### CODE REVIEW

**Review by jspahrsummers — DISMISSED** [review]
Thanks! Can you please make sure all servers are listed in the top-level README.md too?

File: src/aws-kb-retrieval-server/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@modelcontextprotocol/server-aws-kb-retrieval",
+  "version": "0.1.0",
```
[the remark this answers is not in the document]
Comment (Skirano) [reply src/aws-kb-retrieval-server/package.json]: done!

**Review by jspahrsummers — APPROVED** [review]
Thank you!

**Comment by jexp** [comment]
I thought community servers should stay outside of MCP? I'm confused.

**Comment by jspahrsummers** [comment]
@Skirano works with our team directly. However, we are also considering whether to pare down the servers already committed to the repo, to ease maintenance.

**Pull request merged**